src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1579,9 +1645,395 @@ private static Entity tryReturnSource(CommandSource source, CommandArgs args) th
+        return new BigDecimalElement(key);
+    }
+
+    private static class BigDecimalElement extends KeyElement {
+
+        protected BigDecimalElement(Text key) {
+            super(key);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String next = args.next();
+            try {
+                return new BigDecimal(next);
+            } catch (NumberFormatException ex) {
+                throw args.createError(Text.of("Expected a number, but input " + next + " was not"));
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to return a {@link BigInteger}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement bigInteger(Text key) {
+        return new BigIntegerElement(key);
+    }
+
+    private static class BigIntegerElement extends KeyElement {
+
+        protected BigIntegerElement(Text key) {
+            super(key);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String integerString = args.next();
+            try {
+                return new BigInteger(integerString);
+            } catch (NumberFormatException ex) {
+                throw args.createError(Text.of("Expected an integer, but input "+integerString+" was not"));
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to be a valid {@link DataContainer}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement dataContainer(Text key) {
+        return new DataElement(key);
+    }
+
+    private static class DataElement extends RemainingJoinedStringsCommandElement {
+
+        protected DataElement(Text key) {
+            super(key, true);
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String argument = (String) super.parseValue(source, args);
+            Callable<BufferedReader> reader = () -> new BufferedReader(new StringReader(argument));
+            ConfigurationLoader<? extends ConfigurationNode> loader = HoconConfigurationLoader.builder()
+                    .setSource(reader).build();
+            ConfigurationNode node;
+            try {
+                node = loader.load();
+            } catch (IOException ex) {
+                throw args.createError(Text.of("Node parsing failed: ", ex.getMessage()));
+            }
+            return DataTranslators.CONFIGURATION_NODE.translate(node);
+        }
+
+    }
+
+    /**
+     * Expect an argument to be a {@link UUID}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement uuid(Text key) {
```

Comment:
The point here is that it's a plain UUID, not related to anything in particular. `entity()` and `player()` methods already exist should they want it to fit a pattern.